telegram/ext/_application.py

```diff
@@ -78,6 +79,17 @@
 _STOP_SIGNAL = object()
 _DEFAULT_0 = DefaultValue(0)
 
+
+# Since python 3.12, the coroutine passed to create_task should not be an async generator. Remove
+# this check when we drop support for python 3.11.
+if sys.version_info >= (3, 12):
+    _CoroType = Union["asyncio.Future[object]", Awaitable[RT]]
```

Comment:
`asyncio.Future` is not listed in the typeshed repo:

https://github.com/python/typeshed/blob/d492efeaea288177e9eca2734d1d463fbb6fff60/stdlib/asyncio/__init__.pyi#L37-L43

https://github.com/python/typeshed/blob/d492efeaea288177e9eca2734d1d463fbb6fff60/stdlib/asyncio/tasks.pyi#L314-L321

and we also didn't have it before #3543 …

Reply:
Okay, so I replace asyncio.Future with asyncio.Coroutine correct? And we should still leave Awaitable as it is right?

Reply:
My undestanding would be that we don't want `Awaitable` either 🤔

Reply:
so #3540 was incorrectly reported? will there be a typing issue from mypy if we run the example in that issue with asyncio.create_task instead of app.create_task in py 3.12? I can check in a bit

Reply:
> so https://github.com/python-telegram-bot/python-telegram-bot/issues/3540 was incorrectly reported?

maybe. I went through the discussions of #3540 and #3543 a bit more again and I guess the important parts are:

> asyncio.create_task only accepts Coroutines, and Futures are not acceptable:

>  Awaitable is more suitable, considering both asyncio.Future and typing.Coroutine are sub-classes of it.

> That way, Application.create_task would fully cover the functionality of asyncio.gather plus also supporting Future.

That last one is a comment from my side. I would still say that providing a bit more than `asyncio.create_task` doesn't hurt unless it means additional effort for us. If that's the case, we can stick to `Awaitable`. If supporting `Future` does not work easily in py3.12, we'll have to deprecate that as well.

Reply:
Didn't see any issues with also supporting `Future`. So I kept only `Awaitable` since `Coroutine` and `Future` are subclasses of it.